### PR TITLE
#84: Add scalar function capabilities

### LINKED ISSUE #84: Add scalar function capabilities
## Situation

While the SQL renderer is prepared, the adapter does not report the scalar function capabilities.

## Acceptance Criteria

* Supported scalar function capabilities reported by adapter.

### PR DESCRIPTION
**Pull request by redcatbear**
Closes #84.

### CODE REVIEW

File: src/main/lua/exasolvs/query_renderer.lua [hunk 1]
```diff
@@ -83,21 +85,48 @@ function M.new (query)
...
+
     local function append_scalar_function(scalar_function)
         local function_name = string.upper(scalar_function.name)
         if M.supported_scalar_functions[function_name] then
-            append(function_name)
-            if function_name ~= "CURRENT_USER" and function_name ~= "SYSDATE" and function_name ~= "CURRENT_SCHEMA"
-                and function_name ~= "CURRENT_SESSION" and function_name ~= "CURRENT_STATEMENT" then
-                append("(")
+            if function_name == "CURRENT_USER" or function_name == "SYSDATE" or function_name == "CURRENT_SCHEMA"
+                or function_name == "CURRENT_SESSION" or function_name == "CURRENT_STATEMENT" then
```
Comment (kaklakariada) [inline src/main/lua/exasolvs/query_renderer.lua:111]: Extract these five special functions into a list (e.g. `no_arg_functions`?) and check if `function_name` is in this list. That would make the intention clearer.

File: src/test/java/com/exasol/ScalarFunctionsIT.java [hunk 2]
```diff
@@ -14,12 +16,38 @@
...
-        final String sourceSchemaName = "IPROC_SCHEMA";
-        final Schema sourceSchema = createSchema(sourceSchemaName);
+        final Schema sourceSchema = createSchema("IPROC_SCHEMA");
         sourceSchema.createTable("T", "C1", "BOOLEAN").insert(true).insert(false);
         final VirtualSchema virtualSchema = createVirtualSchema(sourceSchema);
-        final User user = createUserWithVirtualSchemaAccess("UP_USER", virtualSchema);
-        assertRlsQueryWithUser("SELECT IPROC() FROM " + getVirtualSchemaName(sourceSchemaName) + ".T", user,
+        final User user = createUserWithVirtualSchemaAccess("IPROC_USER", virtualSchema);
+        assertRlsQueryWithUser("SELECT IPROC() FROM " + getVirtualSchemaName(sourceSchema) + ".T", user,
                 table().row(0).row(0).matches(NO_JAVA_TYPE_CHECK));
```
Comment (kaklakariada) [inline src/test/java/com/exasol/ScalarFunctionsIT.java:24]: `.row(0)` is called twice. What does this mean?

File: src/test/lua/exasolvs/test_query_renderer.lua [hunk 3]
```diff
@@ -570,6 +588,14 @@ end
 
 function test_query_renderer.test_scalar_function_in_select_list_with_two_arguments()
     local parameters = {
+        {
+            func_name = "ADD",
+            first_arg_type = "literal_exactnumeric",
+            first_arg_value = 1,
+            second_arg_type = "literal_exactnumeric",
+            second_arg_value = 1,
+            expected = "SELECT 1 + 1"
```
Comment (kaklakariada) [inline src/test/lua/exasolvs/test_query_renderer.lua:597]:
```suggestion
            first_arg_value = 1,
            second_arg_type = "literal_exactnumeric",
            second_arg_value = 2,
            expected = "SELECT 1 + 2"
```
Use different values to ensure arguments are not mixed up

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: src/main/lua/exasolvs/query_renderer.lua (hunk 1 above)
Comment (redcatbear) [reply src/main/lua/exasolvs/query_renderer.lua:111]: [IPROC]((https://docs.exasol.com/sql_references/functions/alphabeticallistfunctions/iproc.htm) tells you on which cluster node a row resides. We have two rows on a single-node cluster in this test, so the node index is zero for every row.

File: src/test/java/com/exasol/ScalarFunctionsIT.java (hunk 2 above)
Comment (redcatbear) [reply src/test/java/com/exasol/ScalarFunctionsIT.java:24]: [IPROC](https://docs.exasol.com/sql_references/functions/alphabeticallistfunctions/iproc.htm) tells you on which cluster node a row resides. We have two rows on a single-node cluster in this test, so the node index is zero for every row.

File: src/main/lua/exasolvs/query_renderer.lua (hunk 1 above)
Comment (redcatbear) [reply src/main/lua/exasolvs/query_renderer.lua:111]: Extracted method with a speaking name.

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**